src/PruneUnusedContainersModal.jsx

```diff
@@ -0,0 +1,110 @@
+    const [selectedContainerIds, setSelectedContainerIds] = React.useState(unusedContainers.map(u => u.id));
+
+    const handlePruneUnusedContainers = () => {
+        setPruning(true);
+
+        const actions = [];
+
+        for (const id of selectedContainerIds) {
+            if (id.endsWith("true")) {
+                actions.push(client.delContainer(true, id.replace(/true$/, ""), true));
+            } else {
+                actions.push(client.delContainer(false, id.replace(/false$/, ""), true));
+            }
+        }
+        Promise.all(actions).then(close)
+                .catch(ex => {
+                    const error = _("Failed to prune unused containers");
+                    onAddNotification({ type: 'danger', error, errorDetail: ex.message });
+                    close();
+                });
+    };
+
+    const columns = [
+        { title: _("Name"), sortable: true },
+        { title: _("Created"), sortable: true },
+    ];
+
+    if (userSystemServiceAvailable)
+        columns.push({ title: _("Owner"), sortable: true });
+
+    const selectAllContainers = isSelecting => setSelectedContainerIds(isSelecting ? unusedContainers.map(c => c.id) : []);
+    const isContainerSelected = container => selectedContainerIds.includes(container.id);
+    const setContainerSelected = (container, isSelecting) => setSelectedContainerIds(prevSelected => {
+        const otherSelectedContainerNames = prevSelected.filter(r => r !== container.id);
+        return isSelecting ? [...otherSelectedContainerNames, container.id] : otherSelectedContainerNames;
+    });
+
+    const onSelectContainer = (id, _rowIndex, isSelecting) => {
+        const container = unusedContainers.filter(u => u.id === id)[0];
+        setContainerSelected(container, isSelecting);
+    };
+
+    return (
+        <Modal isOpen
+               onClose={close}
+               position="top" variant="medium"
+               title={cockpit.format(_("Prune unused containers"))}
+               footer={<>
+                   <Button id="btn-img-delete" variant="danger"
+                           spinnerAriaValueText={isPruning ? _("Pruning containers") : undefined}
+                           isLoading={isPruning}
+                           isDisabled={selectedContainerIds.length === 0}
+                           onClick={handlePruneUnusedContainers}>
+                       {isPruning ? _("Pruning containers") : _("Prune")}
+                   </Button>
+                   <Button variant="link" onClick={() => close()}>{_("Cancel")}</Button>
+               </>}
+        >
+            <p>{_("Removes selected non-running containers")}</p>
+            <ListingTable columns={columns}
+                          onSelect={(_event, isSelecting, rowIndex, rowData) => onSelectContainer(rowData.props.id, rowIndex, isSelecting)}
+                          onHeaderSelect={(_event, isSelecting) => selectAllContainers(isSelecting)}
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1235-20230517-181252-0ad6985c-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)